pkg/declarativeconfig/transform/permission_set_test.go

```diff
@@ -0,0 +1,57 @@
+package transform
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/declarativeconfig"
+	"github.com/stackrox/rox/pkg/errox"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestWrongConfigurationTypeTransform(t *testing.T) {
+	pt := newPermissionSetTransform()
+	protos, err := pt.Transform(&declarativeconfig.AuthProvider{})
+	assert.Nil(t, protos)
+	assert.Error(t, err)
+	assert.ErrorIs(t, err, errox.InvalidArgs)
+}
+
+func TestTransform(t *testing.T) {
+	permissionSet := &declarativeconfig.PermissionSet{
+		Name:        "some-permission-set",
+		Description: "with a nice description",
+		Resources: []declarativeconfig.ResourceWithAccess{
+			{Resource: "some-resource", Access: declarativeconfig.Access(storage.Access_NO_ACCESS)},
+			{Resource: "another-resource", Access: declarativeconfig.Access(storage.Access_READ_ACCESS)},
+			{Resource: "yet-another-resource", Access: declarativeconfig.Access(storage.Access_READ_WRITE_ACCESS)},
+		},
+	}
+	expectedPermissionSetID := declarativeconfig.NewDeclarativePermissionSetUUID(permissionSet.Name).String()
+	expectedResourceToAccess := map[string]storage.Access{
+		"some-resource":        storage.Access_NO_ACCESS,
+		"another-resource":     storage.Access_READ_ACCESS,
+		"yet-another-resource": storage.Access_READ_WRITE_ACCESS,
+	}
+
+	permissionSetProtoType := reflect.TypeOf((*storage.PermissionSet)(nil))
```

Comment:
```suggestion
	permissionSetProtoType, ok := reflect.TypeOf((*storage.PermissionSet)(nil))
	require.True(t, ok)
```

Reply:
I think you meant to make changes in l.47 when we do `permissionSetProto, ok := protos[permissionSetType][0].(*storage.PermissionSet)`, since this occurrence will be removed with the introduction of the package scoped variable of `reflect.TypeOf((*storage.PermissionSet)(nil)`